exp/api/v1alpha4/awsmanagedmachinepool_types.go

```diff
@@ -83,6 +83,10 @@ type AWSManagedMachinePoolSpec struct {
 	// +optional
 	AdditionalTags infrav1alpha4.Tags `json:"additionalTags,omitempty"`
 
+	// AdditionalASGTags is an optional set of tags to add to the underlying ASG (auto scaling group) resources managed by the AWS provider
+	// +optional
+	AdditionalASGTags infrav1alpha4.Tags `json:"additionalASGTags,omitempty"`
```

Comment:
Do we need the  new `AdditionalASGTags` field or can we use the existing `AdditionalTags`?

Reply:
Is there a scenario where a user (lets say you) where you would want to apply different additional tags to the ASG compared to the EKS control plane?

Reply:
The `AdditionalTags` are applied to the NodeGroup resource, and the `AdditionalASGTags` are intended to be applied to the underlying ASG resource. I'm not sure if we want users to be able to specify them differently (In my case, what I really care about is the tags applied to the ASG for compliance reasons, and they can be the same as the ones applied to the NodeGroup). 

Please let me know what you think. Thanks!

Reply:
In other controllers we use `AdditionalTags` and apply these across different AWS resources.  I think for consistency and to stop us setting a precedent for creating more`AdditionalXXXTags` fields in the future that we should change the  reconciliation of ASG tags to use the existing `AdditionalTags` field. wdyt?

Reply:
That sounds good to me! I'll make the chagne.